packages/wifi_basic/lib/src/extensions.dart

```diff
@@ -1,22 +1,42 @@
 import 'package:wifi_basic/wifi_basic.dart';
 
-extension ToEnumExtension on int? {
-  WiFiGenerations toWifiGeneration() {
-    if (this == null || this! < 0 || this! > 6) {
-      return WiFiGenerations.unknown;
+extension WiFiGenerationsExtension on WiFiGenerations {
```

Comment:
extensions where used as a convince to call methods on integer directly like - `2.toWiFiGenerations`. 

The changed approach looks like a normal function call - in that case I suggest we avoid `extensions` completely - and have a global util function like `_serializeWiFiGeneration`/`_deserializeWiFiGeneration`.

Reply:
Don't you think that we having the (de)serialize logic here is more concise than a global utility method? It's the same concept of using having a `Class.fromJson` and `Class.toJson`.

The way I see, having the fromInt in an extension better gives the message of "this parse should only be used here", different from a **global** function.

It's also better for maintenance, since if we ever rename this enum, we won't need to remember to rename to function.

And last, but not least, for me, the extensions approach makes the code more elegant.